### PR TITLE
Adding SearchBar to Semesters, Courses, Tasks Screen

### PR DESCRIPTION
**Pull request by Jun0se7en**
- Fixed Overlapped Semester Checker
- Adding Search Bar to Semesters, Courses, Tasks Screen
- Viewing Upcoming Course and Task based on Current Semester
- Adding react-native-components package to using SearchBar component

### CODE REVIEW

File: src/components/SearchBars.tsx
Comment (maqnitude) [inline src/components/SearchBars.tsx:1]: Change the tab size to 2.

**Review by maqnitude — CHANGES_REQUESTED** [review]
Sync repo, resolve the conflicts, and link to the appropriate issue(s).

File: src/screens/EditSemesterScreen.tsx [hunk 1]
```diff
@@ -51,8 +51,9 @@ const EditSemesterScreen: React.FC<EditSemesterScreenProps> = ({ navigation, rou
             const existingSemesterStartTime = moment(SemesterDuration.start).format('MM:DD:YYYY');
             const existingSemesterEndTime = moment(SemesterDuration.end).format('MM:DD:YYYY');
 
-            if (((newSemesterStartTime >= existingSemesterStartTime && newSemesterStartTime < existingSemesterEndTime)
-                || (newSemesterEndTime > existingSemesterStartTime && newSemesterEndTime <= existingSemesterEndTime))) {
+            if ((newSemesterStartTime >= existingSemesterStartTime && newSemesterStartTime <= existingSemesterEndTime) ||
+            (newSemesterEndTime >= existingSemesterStartTime && newSemesterEndTime <= existingSemesterEndTime) ||
+            (newSemesterStartTime <= existingSemesterStartTime && newSemesterEndTime >= existingSemesterEndTime)) {
```
Comment (LukasAbraham) [inline src/screens/EditSemesterScreen.tsx:56]: You didn't fix your if statements

File: src/screens/CoursesScreen.tsx [hunk 2]
```diff
@@ -69,15 +72,25 @@ const CoursesScreen: React.FC<CoursesScreenProps> = ({ navigation }) => {
...
   };
 
+  const handleSearch = (searchQuery, courses) => {
+    setSelectedCourses(courses.filter(course => course.name.includes(searchQuery.trim())));
+  };
+
+  useEffect(() => {
+    setSelectedCourses(courses);
+  }, [isLoading, error]);
+
```
Comment (LukasAbraham) [inline src/screens/CoursesScreen.tsx:82]: We don't need another useEffect hook here.

**Review by LukasAbraham — APPROVED** [review]
Your PR still needs to be refactored

**Review by maqnitude — APPROVED** [review]

**Pull request merged**